config.py

```
@@ -63,0 +1,272 @@
+        Example: key_name_parts = ('email', 'smtp_password')
+        """
+        if not isinstance(key_name_parts, (list, tuple)):
+            key_name_parts = [str(key_name_parts)] # Ensure it's iterable
+
+        for part in key_name_parts:
+            part_lower = str(part).lower()
+            if any(s in part_lower for s in SECRET_KEY_SUBSTRINGS):
+                return True
+        return False
+
+    def validate_critical_config(self):
```

Ruff (0.11.9)

`validate_critical_config` is too complex (12 > 10)

(C901)